### PR TITLE
[FIX] web: allow to sign with 4k resolution

### PR DESCRIPTION
**Pull request by arj-odoo**
Before this commit, when a 4K screen was used with devicePixelRatio > 1, the signature svg size was divided by the devicePixelRatio value.

We would have small automatic signature shrinked.

taskid: 3971056

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by fdamhaut — APPROVED** [review]

File: addons/web/static/src/core/signature/name_and_signature.js [hunk 1]
```diff
@@ -245,7 +245,14 @@ export class NameAndSignature extends Component {
      */
     async printImage(imgSrc) {
         this.clear();
-        await this.fromDataURL(imgSrc);
+        // Force the width and height to the canvas size to prevent shrinking on 4K display
+        // with devicePixelRatio > 1
```
Comment (pparidans) [inline addons/web/static/src/core/signature/name_and_signature.js:249]: Applies to all high DPI screens.

File: addons/web/static/src/core/signature/name_and_signature.js [hunk 2]
```diff
@@ -245,7 +245,14 @@ export class NameAndSignature extends Component {
...
+        const options = {
+            width: this.signatureRef.el.width,
+            height: this.signatureRef.el.height,
+            yOffset: 30,
```
Comment (pparidans) [inline addons/web/static/src/core/signature/name_and_signature.js:253]: Not sure I see the point of this offset...

File: addons/web/static/src/core/signature/name_and_signature.scss [hunk 3]
```diff
@@ -6,7 +6,7 @@
 .o_signature_stroke {
     position: absolute;
     border-top: #D1D0CE solid 2px;
-    bottom: 33%;
+    bottom: 20%;
```
Comment (pparidans) [inline addons/web/static/src/core/signature/name_and_signature.scss:9]: Why this change ? Is it related to the `devicePixelRatio` ?

**Review by pparidans — CHANGES_REQUESTED** [review]

File: addons/web/static/src/core/signature/name_and_signature.js (hunk 2 above)
Comment (arj-odoo) [reply addons/web/static/src/core/signature/name_and_signature.js:253]: Indeed, it is not directly related to the issue. It is used to improve the layout. Signature is too high otherwise
image

File: addons/web/static/src/core/signature/name_and_signature.scss (hunk 3 above)
Comment (arj-odoo) [reply addons/web/static/src/core/signature/name_and_signature.scss:9]: Same, as we move the signature with the yOffset, this change move the line too to avoid crossing out the signature. I will move these changes into a separate commit.

**Review by pparidans — APPROVED** [review]
LGTM
@robodoo rebase-ff r+
